website/addons/base/__init__.py

```diff
@@ -527,6 +529,12 @@ def external_accounts(self):
             if x.provider == self.oauth_provider.short_name
         ]
 
+    @oauth_complete.connect
+    def oauth_complete(provider, account, user):
+        if not user or not account:
+            return
+        user.get_or_add_addon(account.provider)
```

Comment:
Also, `user.add_addon` makes sense here, since you're not doing anything with the return value.